data/transactions/logic/assembler.go

```diff
@@ -1226,7 +1227,7 @@ func AssembleStringWithVersionEx(text string, version uint64) ([]byte, map[int]i
 	}
 
 	sr = strings.NewReader(text)
-	ops := OpStream{Version: version}
+	ops := OpStream{Version: version, Stderr: os.Stderr}
```

Comment:
I'm not sure if this one is ok. I don't think that the assembler should write output directly to the stdout/stderr. Instead, it should receive a place to write the error to.
i.e. only the "goal" command line utility should be the one that actually writes to the console. All the rest of the utility functions should return errors / write to provided interfaces, etc.

having said that, I'm ok with deferring this change to a future PR.

Reply:
I agree, I only established the basis needed for tests and it opens options for further improvements.

Reply:
#1710 did a lot to work towards doing this better, probably room for another round after both of these are in